homeassistant/components/sms/sensor.py

```diff
@@ -0,0 +1,78 @@
+"""Support for SMS dongle sensor."""
+import logging
+
+import gammu  # pylint: disable=import-error, no-member
+
+from homeassistant.const import DEVICE_CLASS_SIGNAL_STRENGTH
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, SMS_GATEWAY, SMS_GSM_SIGNAL_SENSOR_NAME
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the GSM Signal Sensor sensor."""
+    name = SMS_GSM_SIGNAL_SENSOR_NAME
+    gateway = hass.data[DOMAIN][SMS_GATEWAY]
+    entities = []
+    entities.append(GSMSignalSensor(hass, gateway, name,))
+    async_add_entities(entities, True)
+
+
+class GSMSignalSensor(Entity):
```

Comment:
Please also include a unique ID property. If you don't have access to a unique identifier, use 

```python
f"{config_entry.entry_id}_signal"
```